src/hw-2/utils.ts

```diff
@@ -0,0 +1,32 @@
+import { User } from './types';
+
+function sortListByLogin(a: User, b: User) {
+    const textA = a.login.toLowerCase();
+    const textB = b.login.toLowerCase();
+
+    if (textA > textB) {
+        return 1;
+    } else if (textA < textB) {
+        return -1;
+    } else {
+        return 0;
+    }
+}
+
+export function getAutoSuggestUsers(loginSubstring: string, limit: number, userList: User[]): User[] {
+    // userList should be sorted if items number in userList are more than one item
+    const newStore = userList.length > 1
+        ? userList.sort(sortListByLogin)
+            .filter((user: User) => user.login.includes(loginSubstring))
+        : userList;
+
+    return limit
+        ? newStore.splice(0, limit - 1)
+        : newStore;
+}
+
+export function getUserIndex(userStore: User[], userId: string): number| null {
+    const userIndex = userStore.findIndex((item: User) => item.id === userId);
+
+    return Boolean(userIndex + 1) ? userIndex: null;
```

Comment:
Strange logic. For making userIndex positive you add plus one and then return it, otherwise just null. Would be easier just `return userStore.findIndex*` without having that strange `number | null`.

Reply:
Fixed. Thanks.